test/xbgpu/test_engine.py

```diff
@@ -107,26 +140,18 @@ def generate_expected_output(
         output_array[..., 1] = 1
         return output_array
     for b in range(batch_start_idx, batch_start_idx + num_batches):
-        sign = pow(-1, b)
         for c in range(channels):
-            h = np.empty((antennas, 2), np.int32)
-            v = np.empty((antennas, 2), np.int32)
+            in_data = np.empty((antennas, 2, 2), np.int32)
```

Comment:
While you're busy tidying up, I'd suggest using `N_POLS` and `COMPLEX` here.

Reply:
Ok, opened #483 for it.